File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/28_find-the-index-of-the-first-occurrence-in-a-string/solutions/cpp/solution.cpp
#include <string>

class Solution {
public:
    int strStr(std::string haystack, std::string needle) {
        if (needle.empty()) return 0; // When needle is empty, return 0 as per problem statement.

        int m = haystack.length();
        int n = needle.length();

        if (m < n) return -1; // Needle is longer than haystack, so it cannot be found.

        for (int i = 0; i <= m - n; ++i) {
            int j = 0;
            while (j < n && haystack[i + j] == needle[j]) {
                j++;
            }
            if (j == n) {
                return i; // Found the needle in the haystack
            }
        }
        return -1; // Needle not found
    }
};

File: DataSets/DataSet-Old/LeetCodeTasksDebug/Fixed/28_find-the-index-of-the-first-occurrence-in-a-string/solutions/python3/solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle == "":
            return 0
        
        # Length of the needle
        needle_length = len(needle)
        
        # Loop through the haystack up to the point where the needle could still fit
        for i in range(len(haystack) - needle_length + 1):
            # Check if the substring from i to i + needle_length is the needle
            if haystack[i:i + needle_length] == needle:
                return i
        return -1
